src/needsahome.jl

```diff
@@ -59,59 +72,61 @@ function buildSubgraphFromLabels(dfg::G,
-- Since this function has happened more than once, it seems the name `buildSubgraphFromLabels!` might stick around, even if it just becomes a wrapper.
-
-Related
-
-buildSubgraphFromLabels!, _copyIntoGraph!, getVariableIds
-"""
-function buildSubgraphFromLabels!_SPECIAL(dfg::G,
-                                          # frontals::Vector{Symbol},
-                                          syms::Vector{Symbol};
-                                          subfg::AbstractDFG=(G <: InMemoryDFGTypes ? G :         GraphsDFG)(params=getSolverParams(dfg)),
-                                          solvable::Int=0,
-                                          allowedFactors::Union{Nothing, Vector{Symbol}}=nothing  )::G  where G <: AbstractDFG
-  #
-  # for sym in separators
-  #   (solvable <= getSolvable(dfg, sym)) && DFG.addVariable!(subfg, deepcopy(DFG.getVariable(dfg, sym)))
-  # end
-
-  addfac = Symbol[]
-  for sym in syms # frontals
-    if solvable <= getSolvable(dfg, sym)
-      addVariable!(subfg, deepcopy(getVariable(dfg, sym)))
-      append!(addfac, getNeighbors(dfg, sym, solvable=solvable))
-    end
-  end
-
-  # allowable factors as intersect between connected an user list
-  usefac = allowedFactors == nothing ? addfac : intersect(allowedFactors, addfac)
-
-  allvars = ls(subfg)
-  for sym in usefac
-    fac = getFactor(dfg, sym)
-    vos = fac._variableOrderSymbols
-    #TODO don't add duplicates to start with
-    if !exists(subfg,fac) && (vos ⊆ allvars) && (solvable <= getSolvable(dfg, sym))
-      addFactor!(subfg, fac._variableOrderSymbols, deepcopy(fac))
-    end
-  end
-
-  # remove orphans
-  for fct in getFactors(subfg)
-    # delete any neighboring factors first
-    if length(getNeighbors(subfg, fct)) != length(fct._variableOrderSymbols)
-      deleteFactor!(subfg, fc)
-      @error "deleteFactor! this should not happen"
-    end
-  end
-
-  return subfg
-end
+## KEEPING COMMENT, WANT TO BE CONSOLIDATED WITH FUNCTION ABOVE -- KEEPING ONLY ONE FOR MAINTAINABILITY
+## STILL NEEDS TO BE CONSOLIDATED WITH `DFG._copyIntoGraph`
+# """
+#     $SIGNATURES
+#
+# IIF clique specific version of building subgraphs.  This is was an unfortunate rewrite of the existing `buildSubgraphFromLabels!` function above.  Currently halfway consolidated.  Tests required to ensure these two functions can be reduced to and will perform the same in both.
+#
+# DevNotes
+# - DF: Could we somehow better consolidate the functionality of this method into `buildSubgraphFromLabels!` above, which in turn should be consolidated as SamC suggests.
+# - Since this function has happened more than once, it seems the name `buildSubgraphFromLabels!` might stick around, even if it just becomes a wrapper.
+#
+# Related
+#
+# buildSubgraphFromLabels!, _copyIntoGraph!, getVariableIds
+# """
+# function buildSubgraphFromLabels!_SPECIAL(dfg::G,
```

Comment:
Just and FYI:  @Affie, this is your add hoc function for subGraph from a while back -- see #250